### PR TITLE
Add function cast to

### PR DESCRIPTION
**Pull request by gregorybarale**

### CODE REVIEW

File: src/castTo/castTo.spec.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+describe('castTo function', () => {
+  const input = { a: 1, b: 2, c: 3 };
+  const array = [{ a: 1, b: 2, c: 3 }, { a: 4, b: 5, c: 6 }];
+
+  test('is a pure function', () => {
+    castTo({ a: x => x })(input);
+    expect(input).toMatchSnapshot();
+  });
+
+  test('{a: 1, b: 2, c:3} |> castTo === {a: 2, b: 2, c: 3}', () => {
```
Comment (evanliomain) [inline src/castTo/castTo.spec.ts]: You miss the purpose of the function in the test.
Please add a test : `'{ a: '1', b: 2.1, c: 'A' } |> castTo(x => parseInt(x.a, 10), x => Math.floor(x.B), x => parseInt(x.c, 16)) === { a: 1, b: 2, c: 10 }'`

File: src/castTo/castTo.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+/**
+ * @module object=>object
+ */
+/**
+ * Cast properties of an object or an array of object, given provided mapping
+ * @param mapping Mapping between key and caster.
+ * @return The function to apply on the object or the array to do cast its properties into another object or array
+ * @example
+ * ```
+ * castTo({a: 1, b: 2, c:3})({a: x => x + x, b: x => x}) // {a: 2, b: 2, c:3}
```
Comment (evanliomain) [inline src/castTo/castTo.ts]: Replace this example by the more meanfull one:
```typescript
castTo(
  ({ a }) => parseInt(a, 10), 
  ({ b }) => Math.floor(b), 
  ({ c }) => parseInt(c, 16)
)({ a: '1', b: 2.1, c: 'A' }) // { a: 1, b: 2, c: 10 }'`

File: src/castTo/castTo.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * castTo([
```
Comment (evanliomain) [inline src/castTo/castTo.ts]: Wrong arguments order

File: src/castTo/castTo.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ *    {a: 1, b: 2, c:3},
+ *    {a: 4, b: 5, c:6}
+ * ])({a: x => x + x, b: x => x}) // [{a: 2, b: 2, c:3}, {a: 8, b: 5, c:6}]
+ * ```
+ * @example Using the chain
+ * ```
+ * chain({a: 1, b: 2, c:3})
```
Comment (evanliomain) [inline src/castTo/castTo.ts]: Replace this example by the more meanfull one:
```typescript
chain({ a: '1', b: 2.1, c: 'A' })
  .chain(castTo(
    ({ a }) => parseInt(a, 10), 
    ({ b }) => Math.floor(b), 
    ({ c }) => parseInt(c, 16)
  ))
  .value() // { a: 1, b: 2, c: 10 }`

File: src/castTo/castTo.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ * ])
+ *   .chain(castTo({a: x => x + x, b: x => x}))
+ *   .value() // [{a: 2, b: 2, c:3}, {a: 8, b: 5, c:6}]
+ * ```
+ */
+export function castTo(mapping: CastMapping): (ob: object) => object {
+  const caster: (ob: object) => object = (ob: object) =>
+    chain(mapping)
+      .chain(entries())
+      .chain(filter(([key, _]) => undefined !== ob[key]))
```
Comment (evanliomain) [inline src/castTo/castTo.ts]: `_` is no usefull here

File: src/castTo/castTo.spec.ts (hunk 1 above)
Comment (gregorybarale) [reply src/castTo/castTo.spec.ts]: Done

File: src/castTo/castTo.ts (hunk 2 above)
Comment (gregorybarale) [reply src/castTo/castTo.ts]: Done

File: src/castTo/castTo.ts (hunk 3 above)
Comment (gregorybarale) [reply src/castTo/castTo.ts]: Done

File: src/castTo/castTo.ts (hunk 4 above)
Comment (gregorybarale) [reply src/castTo/castTo.ts]: Done

File: src/castTo/castTo.ts (hunk 5 above)
Comment (gregorybarale) [reply src/castTo/castTo.ts]: Done

File: src/castTo/__snapshots__/castTo.spec.ts.snap [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    "a": 8,
+    "b": 5,
+    "c": 6,
+  },
+]
+`;
+
+exports[`castTo function { a: '1', b: 2.1, c: 'A' } |> castTo({a: x => parseInt(x.a, 10),b: x => Math.floor(x.B),c: x => parseInt(x.c, 16)}) === { a: 1, b: 2, c: 10 } 1`] = `
+Object {
+  "a": NaN,
```
Comment (evanliomain) [inline src/castTo/__snapshots__/castTo.spec.ts.snap]: Is-it normal to have a NaN here?

Comment (gregorybarale) [reply src/castTo/__snapshots__/castTo.spec.ts.snap]: It's fixed

**Pull request merged**